### PR TITLE
Set TRT network name to builder configs

### PR DESCRIPTION
**Pull request by pooyadavoodi**
This is useful for debugging to tell what builder configs were used when building the engine.

### CODE REVIEW

File: tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc [hunk 1]
```diff
@@ -30,6 +30,7 @@ limitations under the License.
 #include "absl/strings/str_cat.h"
 #include "absl/strings/string_view.h"
 #include "tensorflow/compiler/tf2tensorrt/convert/utils.h"
+#include "tensorflow/compiler/tf2tensorrt/utils/py_utils.h"
```
Comment (aaroey) [inline tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc]: Please avoid depending on this if possible. This is for python only, it depends on TF's dso_loader, and I remembered it has caused some problems before (though I forgot how exactly).

Instead, please add new methods Get{Linked,Loaded}TensorRTVersion and maybe let them return a string to make it more convenient. We already duplicate the code in line ~1189.

Thanks.

File: tensorflow/compiler/tf2tensorrt/kernels/trt_engine_op.cc [hunk 2]
```diff
@@ -519,6 +519,20 @@ bool TRTEngineOp::ExecuteTrtEngine(OpKernelContext* ctx,
                                    EngineContext* engine_context) {
   VLOG(1) << "Executing TRT engine: " << name();
   auto& cuda_engine = engine_context->cuda_engine;
+
+  if (VLOG_IS_ON(2)) {
+    VLOG(2) << "  Network name: " << cuda_engine->getName();
```
Comment (aaroey) [inline tensorflow/compiler/tf2tensorrt/kernels/trt_engine_op.cc]: Can we make sure that the TRTEngineOp's setting matches the network name?

**Review by aaroey — CHANGES_REQUESTED** [review]

File: tensorflow/compiler/tf2tensorrt/kernels/trt_engine_op.cc (hunk 2 above)
Comment (pooyadavoodi) [reply tensorflow/compiler/tf2tensorrt/kernels/trt_engine_op.cc]: I think this is a good idea, but I am concerned if it adds to the latency of inference.
To do this, we would have to do the following at every inference step for each of the 4 parameters precision_mode, use_calibration, max_batch_size, max_workspace_size:
- Do a pattern matching in the network name to find the parameter value as a substring.
- Convert the TRTEngineOp parameter to a string
- Compare the found pattern with the converted parameter

File: tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc (hunk 1 above)
Comment (pooyadavoodi) [reply tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc]: Done. Could you please review if I added the headerfiles and namespaces properly?

File: tensorflow/compiler/tf2tensorrt/convert/utils.cc [hunk 3]
```diff
@@ -51,5 +51,42 @@ Status TrtPrecisionModeFromName(const string& name, TrtPrecisionMode* mode) {
...
+#if GOOGLE_CUDA && GOOGLE_TENSORRT
+  major = NV_TENSORRT_MAJOR;
+  minor = NV_TENSORRT_MINOR;
+  patch = NV_TENSORRT_PATCH;
+#else
+  major = 0;
+  minor = 0;
+  patch = 0;
+#endif
+  string trt_version = std::to_string(major) + "." +
```
Comment (aaroey) [inline tensorflow/compiler/tf2tensorrt/convert/utils.cc]: StrCat here and below as well?

**Review by aaroey — CHANGES_REQUESTED** [review]
Thanks for the fix! Just some minor comment and we're good to go.

File: tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc [hunk 4]
```diff
@@ -1391,6 +1389,20 @@ Status Converter::BuildCudaEngine(
     }
   }
 
+  string precision_mode_str;
+  TF_RETURN_IF_ERROR(TrtPrecisionModeToName(
+      precision_mode_, &precision_mode_str));
+  string trt_version_str = GetLoadedTensorRTVersion();
+  string trt_network_name =
+      "TF" + string(TF_VERSION_STRING) + "-" +
```
[the remark this answers is not in the document]
Comment (pooyadavoodi) [reply tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc]: Done

File: tensorflow/compiler/tf2tensorrt/convert/utils.cc (hunk 3 above)
Comment (pooyadavoodi) [reply tensorflow/compiler/tf2tensorrt/convert/utils.cc]: Done

File: tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc [hunk 5]
```diff
@@ -1393,13 +1393,13 @@ Status Converter::BuildCudaEngine(
...
   string trt_version_str = GetLoadedTensorRTVersion();
-  string trt_network_name =
-      "TF" + string(TF_VERSION_STRING) + "-" +
-      "TRT" + trt_version_str + "-" +
-      "Precision-" + precision_mode_str + "-" +
-      "Calibration-" + std::to_string(use_calibration_) + "-" +
-      "Max-Batch-Size-" + std::to_string(max_batch_size) + "-" +
-      "Max-Workspace-Size-" + std::to_string(max_workspace_size_bytes);
+  string trt_network_name = StrCat(
+      "TF:", string(TF_VERSION_STRING), ", ",
```
Comment (aaroey) [inline tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc]: nit: please remove calls to string() and to_string(). There is no need to construct a string explicitly with StrCat.

File: tensorflow/compiler/tf2tensorrt/convert/utils.cc [hunk 6]
```diff
@@ -62,9 +63,9 @@ string GetLinkedTensorRTVersion() {
   minor = 0;
   patch = 0;
 #endif
-  string trt_version = std::to_string(major) + "." +
-                       std::to_string(minor) + "." +
-                       std::to_string(patch);
+  string trt_version = absl::StrCat(std::to_string(major), ".",
```
Comment (aaroey) [inline tensorflow/compiler/tf2tensorrt/convert/utils.cc]: Same here and below

**Review by aaroey — CHANGES_REQUESTED** [review]

File: tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc (hunk 5 above)
Comment (pooyadavoodi) [reply tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc]: Done

File: tensorflow/compiler/tf2tensorrt/convert/utils.cc (hunk 6 above)
Comment (pooyadavoodi) [reply tensorflow/compiler/tf2tensorrt/convert/utils.cc]: Done

**Review by aaroey — DISMISSED** [review]

**Comment by aaroey** [comment]
@pooyadavoodi could you please help to resolve the conflict, and make sure all the tests are passed at head with this PR? Thanks.

**Comment by pooyadavoodi** [comment]
> @pooyadavoodi could you please help to resolve the conflict, and make sure all the tests are passed at head with this PR? Thanks.

I resolved the conflicts and ran all the tests. Thanks @aaroey

**Review by aaroey — DISMISSED** [review]

**Review by aaroey — APPROVED** [review]

**Pull request merged**